tests/Integration/Webservice/AddressesTest.php

```diff
@@ -0,0 +1,315 @@
+        $data = '<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
+            <address>
+                <id><![CDATA[' . $id . ']]></id>
+                <id_customer><![CDATA[0]]></id_customer>
+                <id_manufacturer><![CDATA[0]]></id_manufacturer>
+                <id_supplier><![CDATA[0]]></id_supplier>
+                <id_warehouse><![CDATA[0]]></id_warehouse>
+                <id_country xlink:href="http://127.0.0.1/PrestaShop/api/countries/21"><![CDATA[21]]></id_country>
+                <id_state xlink:href="http://127.0.0.1/PrestaShop/api/states/35"><![CDATA[35]]></id_state>
+                <alias><![CDATA[supplier]]></alias>
+                <company><![CDATA[Fashion]]></company>
+                <lastname><![CDATA[supplier]]></lastname>
+                <firstname><![CDATA[supplier]]></firstname>
+                <vat_number><![CDATA[]]></vat_number>
+                <address1><![CDATA[767 Fifth Ave.]]></address1>
+                <address2><![CDATA[]]></address2>
+                <postcode><![CDATA[10153]]></postcode>
+                <city><![CDATA[Paris]]></city>
+                <other><![CDATA[]]></other>
+                <phone><![CDATA[+33 1 23 45 67 89]]></phone>
+                <phone_mobile><![CDATA[+33 6 89 67 45 23]]></phone_mobile>
+                <dni><![CDATA[]]></dni>
+                <deleted><![CDATA[0]]></deleted>
+            </address>
+        </prestashop>';
+        $expected = new Crawler();
+        $expected->addXmlContent($data);
+
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'PUT', 'addresses/' . $id, $data);
+
+        // Check values from the input data
+        $children = $output->filter('address')->children();
+        foreach ($children as $item) {
+            /** @var \DOMElement $item */
+            $property = $item->nodeName;
+            /** @var \DOMElement $expectedItem */
+            $expectedItem = $expected->filter('prestashop > address >' . $property)->getNode(0);
+            if ($property == 'id') {
+                self::assertNotEmpty($item->nodeValue);
+                continue;
+            }
+            if ($property == 'date_add' || $property == 'date_upd') {
+                continue;
+            }
+            self::assertEquals(
+                $expectedItem->nodeValue,
+                $item->nodeValue,
+                sprintf(
+                    'In field %s, the expected value "%s" is not equals to "%s"',
+                    $property,
+                    $expectedItem->nodeValue,
+                    $item->nodeValue
+                )
+            );
+        }
+
+        // Check count
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'GET', 'addresses');
+        self::assertEquals(self::INITIAL_COUNT + 1, $output->filter('prestashop > addresses > address')->count());
+
+        // Delete object in database
+        $output = $this->requestWebserviceXML($this->wsKey->key, 'DELETE', 'addresses/' . $id);
```

Comment:
suggestion : Can we use constants like Symfony\Component\HttpFoundation\Request::METHOD_DELETE for HTTP verbs ?